bundles/org.openhab.binding.mideaac/src/main/java/org/openhab/binding/mideaac/internal/handler/MideaACHandler.java

```diff
@@ -0,0 +1,1501 @@
+            if (connectionMessage) {
+                logger.info("Connection issue, resetting, please wait ...");
+            }
+            connectionMessage = false;
+            getConnectionManager().cancelConnectionMonitorJob();
+            getConnectionManager().disconnect();
+            getConnectionManager().scheduleConnectionMonitorJob();
+        }
+    }
+
+    private boolean isOnline() {
+        return thing.getStatus().equals(ThingStatus.ONLINE);
+    }
+
+    private boolean isOffline() {
+        return thing.getStatus().equals(ThingStatus.OFFLINE);
+    }
+
+    /**
+     * Cancel the connection manager job which will keep going
+     * even with the binding removed and cause warnings about
+     * trying to update Thing Channels with the Handler disposed
+     */
+    @Override
+    public void dispose() {
+        connectionManager.cancelConnectionMonitorJob();
+        markOffline();
+    }
+
+    /**
+     * DoPoll is set to false in the MideaAC Handler
+     * if a Command is being sent and picked up by
+     * the Connection Manager. Then is reset to true
+     * after the Set command is complete
+     * 
+     * @return doPoll Sets if the binding will poll after authorization
+     */
+    public boolean getDoPoll() {
+        return doPoll;
+    }
+
+    /**
+     * Resets the doPoll switch
+     */
+    public void resetDoPoll() {
+        doPoll = true;
+    }
+
+    /**
+     * Reset Retry controls the short 5 second delay
+     * Before starting 30 second delays. (More severe Wifi issue)
+     * It is reset after a successful connection
+     */
+    public void resetRetry() {
+        retry = true;
+    }
+
+    /**
+     * Limit logging of INFO connection messages to
+     * only when the device was Offline in its prior
+     * state
+     */
+    public void resetConnectionMessage() {
+        connectionMessage = true;
+    }
+
+    private ThingStatusDetail getDetail() {
+        return thing.getStatusInfo().getStatusDetail();
+    }
+
+    /**
+     * Sets Cloud Provider
+     * 
+     * @param cloudProvider Cloud Provider
+     */
+    public void setCloudProvider(CloudProvider cloudProvider) {
+        this.cloudProvider = cloudProvider;
+    }
+
+    /**
+     * Security methods
+     * 
+     * @param security security class
+     */
+    public void setSecurity(Security security) {
+        this.security = security;
+    }
+
+    /**
+     * The {@link ConnectionManager} class is responsible for managing the state of the TCP connection to the
+     * indoor AC unit evaporator.
+     *
+     * @author Jacek Dobrowolski - Initial Contribution
+     * 
+     * @author Bob Eckhoff - Revised logic to reconnect with security before each poll or command
+     * 
+     *         This gets around the issue that any command needs to be within 30 seconds of the authorization
+     *         in testing this only adds 50 ms, but allows polls at longer intervals
+     */
+    public class ConnectionManager {
```

Comment:
Please extract this class to its own file

Reply:
Creating a separate class for the ConnectionManager is counter to my guideline to not fix what isn’t broke. I set the guideline recognizing my skill level and not being the original author. I did take a look and there are lots of interactions between MideaACHandler and the ConnectionManager that would need to be addressed just to get it back to working like it is now.  I did change the ConnectionManager back to a private class.  It was private originally, but I changed it when trying to write a test, but forgot to change it back.

Reply:
I would like an example of this suggestion `Usually this ConnectionManager informs the caller by using custom exceptions like AuthenticationException or TransientNetworkErrorException when one of the public methods is unable to return a value or perform a coimmand. The caller (in this case the Handler) can route that to updateStatus() easily. `to copy.  Say on this current code segment.  Both the configuration update and updatStatus seem problematic.
```
       private void getTokenKeyCloud(CloudProviderDTO cloudProvider) {
            CloudDTO cloud = mideaACHandler.getClouds().get(config.email, config.password, cloudProvider);
            if (cloud != null) {
                cloud.setHttpClient(httpClient);
                if (cloud.login()) {
                    TokenKey tk = cloud.getToken(config.deviceId);
                    Configuration configuration = editConfiguration();

                    configuration.put(CONFIG_TOKEN, tk.token());
                    configuration.put(CONFIG_KEY, tk.key());
                    updateConfiguration(configuration);

                    logger.trace("Token: {}", tk.token());
                    logger.trace("Key: {}", tk.key());
                    logger.info("Token and Key obtained from cloud, saving, initializing");
                    initialize();
                } else {
                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, String.format(
                            "Can't retrieve Token and Key from Cloud; email, password and/or cloud parameter error"));
                    logger.warn(
                            "Can't retrieve Token and Key from Cloud; email, password and/or cloud parameter error");
                }
            }
        }
```

Reply:
I'll fix this for you, give me a moment to show you how this could be done.

Reply:
Thanks! Also after some thought was wondering if this was a bad example (should be pulled back into the handler), but I really just need an example to get going.

Reply:
Seems a bit more work then expected, all this is tangled together. Also don;t have too much time today, will continue and probably finish before monday. I'll create a PR on your repo so you can inspect and decide what to do.

Reply:
Again thanks in advance for whatever you can provide to me started.

Reply:
Added a PR to your branch. There you can inspect, reject, merge or ask for changes. 
https://github.com/apella12/openhab-addons/pull/1